content/blog/end-to-end-easy-to-use-pipeline-for-training-a-model-on-medmnist-v2-using-hpe-machine-learning-development-environment-flask.md

```diff
@@ -0,0 +1,514 @@
+
+    # Add a batch dimension
+    processed_data = processed_data.unsqueeze(0)
+
+    return processed_data
+```
+
+Here, we perform the same normalizations done when training the model on the training dataset.  
+
+#### Step 2.3: Define a Flask server  
+
+```python
+# Define a Flask route for serving predictions
+@app.route("/predict", methods=["POST"])
+def predict():
+    # Get the input image file from the request
+    file = request.files["file"]
+
+    # Read the image file into a PIL Image object
+    image = Image.open(io.BytesIO(file.read()))
+
+    # Preprocess the input data
+    processed_data = preprocess_data(image)
+
+    # Use the Determined model to make a prediction
+    output_tensor = model(processed_data)
+
+    # Convert the output tensor to a numpy array
+    output_array = output_tensor.detach().numpy()
+
+    # Apply softmax to get probabilities for each class
+    probabilities = F.softmax(output_tensor, dim=1)
+
+    # Convert the output tensor to a numpy array
+    output_array = probabilities.detach().numpy()
+
+    # Get the predicted class label
+    class_label = np.argmax(output_array)
+    class_labels = [
+        "adipose",
+        "background",
+        "debris",
+        "lymphocytes",
+        "mucus",
+        "smooth muscle",
+        "normal colon mucosa",
+        "cancer-associated stroma",
+        "colorectal adenocarcinoma epithelium",
+    ]
+
+    # Return the predicted class label as a JSON response
+    return jsonify({"prediction": class_labels[class_label]})
+```
+
+Here we define a server that accepts an image file, performs the preprocessing steps defined in the preprocess_data function, and outputs the prediction as a json object.  
+
+#### Step 2.4: Define a function to start the Flask server: 
+
+```python
+# Start the Flask server
+if __name__ == "__main__":
+    app.run(debug=True, host="0.0.0.0", port=5000)
+```
+
+In this example the server is deployed on localhost on port 5000. 
+
+#### Step 2.5: Obtain test data
+
+Your test data should be in the form of .jpeg, .png, or .jpg images, deploy the Flask server, and submit requests for inference 
+
+Full instructions for running the example can be found in the [repository](https://github.com/ighodgao/determined_medmnist_e2e). 
+
+For example, running prediction on the following image of adipose tissue results in: 
+
+![](/img/adipose.jpeg)
+
+`"prediction": "adipose"`
+
+Deploying your model can make it a lot easier to use in production, and while this example uses Flask as a simple proof-of-concept, tools like KServe can help deploy your models in production. 
+
+### Conclusion  
+
+That’s it! We have shown how to implement an end-to-end machine learning workflow using HPE Machine Learning Development Environment and Flask. For more information about HPE’s AI toolkit please visit the [HPE AI Solutions homepage here.](https://www.hpe.com/us/en/solutions/artificial-intelligence.html)  
```

Comment:
Second sentence, change to:
"I've shown you how to implement..."
Last sentence, add a comma prior to "please visit"